tools/update_sample_data.py

```diff
@@ -0,0 +1,102 @@
+# /// script
+# requires-python = ">=3.11"
+# dependencies = [
+#   "numpy",
+#   "astropy",
+#   "sunpy[net]",
+#   "dkist",
+# ]
+# ///
+"""
+This script recreates the sample data files and uploads the recreated versions to Asgard.
+"""
+import sys
+import tarfile
+import argparse
+from pathlib import Path
+
+import numpy as np
+
+from sunpy.net import Fido
+from sunpy.net import attrs as a
+
+import dkist
+import dkist.net
+from dkist.net.globus import start_transfer_from_file_list, watch_transfer_progress
+from dkist.net.globus.endpoints import get_local_endpoint_id, get_transfer_client
+
+datasets = {
+    "AJQWW": {
+        "tiled": True,
+        "tile_slice": np.s_[0],
+        "filename": "AJQWW_single_mosaic.tar",
+    },
+    "BKPLX": {
+        "tiled": False,
+        "slice": np.s_[0],
+        "filename": "BKPLX_stokesI.tar",
+    },
+}
+
+def main(datasets, working_directory, destination_path="/user_tools_tutorial_data/"):
+    working_directory = Path(working_directory)
+    working_directory.mkdir(parents=True, exist_ok=True)
+    sample_files_for_upload = []
+
+    for did, props in datasets.items():
+        res = Fido.search(a.dkist.Dataset(did))
+        asdf_file = Fido.fetch(res, path=working_directory / "{dataset_id}", progress=False, overwrite=False)
+
+        ds = dkist.load_dataset(asdf_file)
+        if "slice" in props:
+            ds = ds[props["slice"]]
+        if "tile_slice" in props:
+            ds = ds.slice_tiles[props["tile_slice"]]
```

Comment:
Do we need separate variables here? We could just do the indexing in the if tiled/else block.

Reply:
Well you could also slice down how many tiles you have like this with `slice`.

Reply:
Yeah, the idea here is to allow this to scale to other future datasets we might want to include, for example if we have a DL one we might only want to ship a subset of the tiles for a moasic.